### PR TITLE
importccl: fix IMPORT publishTables() bug

### PR DESCRIPTION
**Pull request by adityamaru**
Informs: #46729

### CODE REVIEW

File: pkg/ccl/importccl/import_stmt_test.go [hunk 1]
```diff
@@ -3651,6 +3652,83 @@ func TestImportControlJob(t *testing.T) {
 	})
 }
 
+// TestImportCancelAfterCompletion ensures that a canceled IMPORT job cleans up
+// even after the tables have been published i.e the job has been "completed".
+func TestImportCancelAfterCompletion(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	skip.UnderRace(t, "node liveness error")
```
Comment (adityamaru) [inline pkg/ccl/importccl/import_stmt_test.go:3661]: this test is bound to flake with `node liveness error`. Is there a way to prevent this?

**Review by pbardea — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52604)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru, @miretskiy, and @pbardea)

___
*[pkg/ccl/importccl/import_stmt_test.go, line 3661 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/52604#-MEQ-I-qh2FO76zzrmRc-r2-3661:-MET2DTRDH1DFYNg_Oej:byrlfda) ([raw file](https://github.com/cockroachdb/cockroach/blob/f33cc5957a6656562a3f7a6933f4809bfe67d93c/pkg/ccl/importccl/import_stmt_test.go#L3661)):*
<details><summary><i>Previously, adityamaru (Aditya Maru) wrote…</i></summary><blockquote>

this test is bound to flake with `node liveness error`. Is there a way to prevent this?
</blockquote></details>

I wonder if there's something that's bringing the cluster to a halt... I wonder if starting a 1-node test cluster would help with this?

**Comment by miretskiy** [comment]
Have you seen `pkg/ccl/importccl/import_processor_test.go:649:
            RegistryLiveness: jobs.NewFakeNodeLiveness(1),`

**Review by ajwerner — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52604)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru and @miretskiy)

___
*[pkg/ccl/importccl/import_stmt_test.go, line 3661 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/52604#-MEQ-I-qh2FO76zzrmRc-r2-3661:-MYG4v4--xl03F7NKgUy:b-htz0d) ([raw file](https://github.com/cockroachdb/cockroach/blob/f33cc5957a6656562a3f7a6933f4809bfe67d93c/pkg/ccl/importccl/import_stmt_test.go#L3661)):*
<details><summary><i>Previously, pbardea (Paul Bardea) wrote…</i></summary><blockquote>

I wonder if there's something that's bringing the cluster to a halt... I wonder if starting a 1-node test cluster would help with this?
</blockquote></details>

FWIW jobs don't use node liveness anymore...

**Pull request closed without merging**